tensorflow_addons/optimizers/conditional_gradient.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Conditional Gradient method for TensorFlow."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.utils import keras_utils
+
+
+@keras_utils.register_keras_custom_object
+class ConditionalGradient(tf.keras.optimizers.Optimizer):
+    """Optimizer that implements the Conditional Gradient optimization.
+    Helps handle constraints well.
+    Currently only supports frobenius norm constraint.
+    See https://arxiv.org/pdf/1803.06453.pdf
+    ```
+    variable -= (1-learning_rate)
+        * (variable + lamda * gradient / frobenius_norm(gradient))
+    ```
+    """
+
+    def __init__(self,
+                 learning_rate,
+                 lamda,
+                 use_locking=False,
+                 name="ConditionalGradient",
+                 **kwargs):
+        """Construct a conditional gradient optimizer.
+            Args:
+            learning_rate: A `Tensor` or a floating point value.
+                           The learning rate.
+            lamda: A `Tensor` or a floating point value. The constraint.
+            use_locking: If `True` use locks for update operations.
+            name: Optional name prefix for the operations created when
+                  applying gradients.  Defaults to "ConditionalGradient"
+        """
+        super(ConditionalGradient, self).__init__(name=name, **kwargs)
+        self._set_hyper("learning_rate", learning_rate)
+        self._set_hyper("lamda", lamda)
+        self._set_hyper("use_locking", use_locking)
+
+    def get_config(self):
+        config = {
+            'learning_rate': self._learning_rate,
+            'lamda': self._lamda,
+            'use_locking': self._use_locking
+        }
+        base_config = super(ConditionalGradient, self).get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+    def _create_slots(self, var_list):
+        for v in var_list:
+            self.add_slot(v, "conditional_gradient")
+
+    def _prepare(self, var_list):
+        learning_rate = self.learning_rate
```

Comment:
`self.learning_rate` isn't set anywhere. (same for `self.lamda`)

I might be mistaken, but using `self._set_hyper("learning_rate", learning_rate)` doesn't create a new attr `self.learning_rate`. 

Here's the definition of `_set_hyper`:
```python
  def _set_hyper(self, name, value):
    """set hyper `name` to value. value can be callable, tensor, numeric."""
    if isinstance(value, trackable.Trackable):
      self._track_trackable(value, name, overwrite=True)
    if name not in self._hyper:
      self._hyper[name] = value
    else:
      prev_value = self._hyper[name]
      if (callable(prev_value)
          or isinstance(prev_value,
                        (ops.Tensor, int, float,
                         learning_rate_schedule.LearningRateSchedule))
          or isinstance(value, learning_rate_schedule.LearningRateSchedule)):
        self._hyper[name] = value
      else:
        backend.set_value(self._hyper[name], value)
```

You should instead be using `self._get_hyper('learning_rate')`